monkey/monkey_island/cc/services/post_breach_files.py

```diff
@@ -1,53 +1,44 @@
 import logging
 import os
-
-import monkey_island.cc.services.config
+from pathlib import Path
 
 __author__ = "VakarisZ"
 
-import monkey_island.cc.environment.environment_singleton as env_singleton
-
 logger = logging.getLogger(__name__)
 
 # Where to find file names in config
-PBA_WINDOWS_FILENAME_PATH = ["monkey", "post_breach", "PBA_windows_filename"]
 PBA_LINUX_FILENAME_PATH = ["monkey", "post_breach", "PBA_linux_filename"]
+PBA_WINDOWS_FILENAME_PATH = ["monkey", "post_breach", "PBA_windows_filename"]
 
 
-def remove_PBA_files():
-    if monkey_island.cc.services.config.ConfigService.get_config():
-        windows_filename = monkey_island.cc.services.config.ConfigService.get_config_value(
-            PBA_WINDOWS_FILENAME_PATH
-        )
-        linux_filename = monkey_island.cc.services.config.ConfigService.get_config_value(
-            PBA_LINUX_FILENAME_PATH
-        )
-        if linux_filename:
-            remove_file(linux_filename)
-        if windows_filename:
-            remove_file(windows_filename)
-
-
-def remove_file(file_name):
-    file_path = os.path.join(env_singleton.env.get_config().data_dir_abs_path, file_name)
-    try:
-        if os.path.exists(file_path):
-            os.remove(file_path)
-    except OSError as e:
-        logger.error("Can't remove previously uploaded post breach files: %s" % e)
-
-
-def set_config_PBA_files(config_json):
-    """
-    Sets PBA file info in config_json to current config's PBA file info values.
-    :param config_json: config_json that will be modified
-    """
-    if monkey_island.cc.services.config.ConfigService.get_config():
-        linux_filename = monkey_island.cc.services.config.ConfigService.get_config_value(
-            PBA_LINUX_FILENAME_PATH
-        )
-        windows_filename = monkey_island.cc.services.config.ConfigService.get_config_value(
-            PBA_WINDOWS_FILENAME_PATH
+class PostBreachFilesService:
+    DATA_DIR = None
+    CUSTOM_PBA_DIRNAME = "custom_pbas"
+
+    # TODO: A number of these services should be instance objects instead of
+    # static/singleton hybrids. At the moment, this requires invasive refactoring that's
+    # not a priority.
+    @classmethod
+    def initialize(cls, data_dir):
+        cls.DATA_DIR = data_dir
```

Comment:
In general, global objects should be immutable (as in you shouldn't change the value during the runtime). If the `data_dir` is already in the singleton, why do we need to create another global object `PostBreachFileService.DATA_DIR` to use the same value?